src/transformers/modeling_utils.py

```diff
@@ -2280,7 +2288,7 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                 loaded_state_dict_keys = sharded_metadata["all_checkpoint_keys"]
             else:
                 loaded_state_dict_keys = [k for k in state_dict.keys()]
-            if low_cpu_mem_usage:
+            if low_cpu_mem_usage or cls._keep_in_fp32_modules is not None:
```

Comment:
I don't really like this solution, it forces the user to have `accelerate` installed if they want to use t5, even for a small model. Note that the original issue comes only for large models. So it's fine to not upcast those weights in fp32 for small models. Proposed an alternative in [a62a594](https://github.com/huggingface/transformers/pull/20683/commits/a62a594e6a97ce32ed5b070ff894dfe2b54af0af)